khalid-lab/server.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const ee = new EE();
+const Client = require('./model/client.js');
+const PORT = process.env.PORT || 3000;
+const server = net.createServer();
+const clientsArray = [];
+
+server.on('connection', function(socket){
+  var client = new Client(socket);
+  client.socket.write('++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n');
+  client.socket.write('+            WELCOME TO KHALIDLAND                     +\n');
+  client.socket.write('++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n');
+  clientsArray.push(client);
+  console.log('USER CONNECTED!!!');
+  // socket.on('error', function(err){
```

Comment:
An error event won't fire unless there actually is an error so the conditional you are using is not actually necessary. Node is already handling that with the error event. 👍